pkg/cmd/env/options.go

```diff
@@ -50,11 +53,19 @@ func (o *options) Complete(f util.Factory, cmd *cobra.Command, args []string) er
 	o.GardenDir = f.GardenHomeDir()
 	o.Template = newTemplate("usage-hint")
 
-	if o.ProviderType == "kubernetes" {
+	switch o.ProviderType {
+	case "kubernetes":
 		filename := filepath.Join(o.GardenDir, "templates", "kubernetes.tmpl")
 		if err := o.Template.ParseFiles(filename); err != nil {
 			return err
 		}
+	case "azure", "gcp":
+		manager, err := f.Manager()
+		if err != nil {
+			return err
+		}
+
+		o.SessionDir = manager.SessionDir()
```

Comment:
hm, I'm wondering if it wouldn't be better to split the options for each use case: kubectl-env vs provider-env

sessionDir would only be set on the provider-env options. This would make it clearer when which option is available (without having to clarify it in the go-doc). Also we would get rid of the switches, not sure if there is a lot of code that

If we would agree on that, I'm also fine to postpone this refactoring and do it in a separate PR later on. WDYT?

Reply:
I agree and will create an issue and make that change in a separate PR.

Reply:
#76